packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -53,23 +56,34 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
 
     deinitialize(): Promise<void> {
         this._isCredit = undefined;
+        this._isVenmo = undefined;
 
         return Promise.resolve();
     }
 
     private _handleClickButtonProvider({ fundingSource }: ClickDataOptions): void {
         this._isCredit = fundingSource === 'credit' || fundingSource === 'paylater';
+        this._isVenmo = fundingSource === 'venmo';
     }
 
     private _tokenizePayment({ orderID }: ApproveDataOptions) {
         if (!orderID) {
             throw new MissingDataError(MissingDataErrorType.MissingPayment);
         }
+        let provider;
+
+        if (this._isVenmo && this._isVenmoEnabled) {
```

Comment:
Checked with @serhii-tkachenko and we are going to split this strategy into their respective ones going forward so this seems ok for now.